orm/serial_model_bucket_test.go

```diff
@@ -0,0 +1,473 @@
+	assert.Nil(t, cnt.GetID())
+	if err := b.Upsert(db, &cnt); !errors.ErrModel.Is(err) {
+		t.Fatalf("want a model error, got %+v", err)
+	}
+
+	// Using a key should not throw error
+	cnt = CounterWithID{ID: weavetest.SequenceID(1), Count: 111}
+	err := b.Upsert(db, &cnt)
+	assert.Nil(t, err)
+
+	// Get from db and compare
+	var loaded CounterWithID
+	err = b.One(db, weavetest.SequenceID(1), &loaded)
+	assert.Nil(t, err)
+	assert.Equal(t, weavetest.SequenceID(1), loaded.GetID())
+	assert.Equal(t, int64(111), loaded.Count)
+
+	// Overwrite saved model
+	cnt = CounterWithID{ID: weavetest.SequenceID(1), Count: 333}
+	err = b.Upsert(db, &cnt)
+	assert.Nil(t, err)
+
+	// Get from db and compare
+	err = b.One(db, weavetest.SequenceID(1), &loaded)
+	assert.Nil(t, err)
+	assert.Equal(t, weavetest.SequenceID(1), loaded.GetID())
+	assert.Equal(t, int64(333), loaded.Count)
+}
+
+func TestSerialModelBucketPrefixScan(t *testing.T) {
+	db := store.MemStore()
+
+	b := NewSerialModelBucket("cnts", &CounterWithID{})
+
+	cnts := []CounterWithID{
+		CounterWithID{Count: 1},
+		CounterWithID{Count: 17},
+		CounterWithID{Count: 11},
+		CounterWithID{Count: 3},
+	}
+
+	for i := range cnts {
+		// make sure we point to value in array, so this ID gets set
+		err := b.Create(db, &cnts[i])
+		assert.Nil(t, err)
+	}
+
+	var loaded CounterWithID
+	iter, err := b.PrefixScan(db, nil, false)
+	assert.Nil(t, err)
+
+	err = iter.LoadNext(&loaded)
+	assert.Nil(t, err)
+	assert.Equal(t, cnts[0], loaded)
+
+	err = iter.LoadNext(&loaded)
+	assert.Nil(t, err)
+	assert.Equal(t, cnts[1], loaded)
+
+	iter.Release()
+
+	// validate reverse also works
+	iter, err = b.PrefixScan(db, nil, true)
+	assert.Nil(t, err)
+	err = iter.LoadNext(&loaded)
+	assert.Nil(t, err)
+	assert.Equal(t, cnts[len(cnts)-1], loaded)
+	iter.Release()
+}
+
+func TestSerialModelBucketPrefixIteratorDone(t *testing.T) {
+	db := store.MemStore()
+
+	b := NewSerialModelBucket("cnts", &CounterWithID{})
+
+	cntr := CounterWithID{Count: 1}
+	// make sure we point to value in array, so this ID gets set
+	err := b.Create(db, &cntr)
+	assert.Nil(t, err)
+
+	var loaded CounterWithID
+	iter, err := b.PrefixScan(db, nil, false)
+	assert.Nil(t, err)
+
+	err = iter.LoadNext(&loaded)
+	assert.Nil(t, err)
+
+	if err = iter.LoadNext(&loaded); !errors.ErrIteratorDone.Is(err) {
+		t.Fatalf("unexpected error: %s", err)
+	}
+}
+
+func lexographicCountIndex(obj Object) ([]byte, error) {
+	c, ok := obj.Value().(*CounterWithID)
+	if !ok {
+		return nil, errors.Wrapf(errors.ErrType, "%T", obj.Value())
+	}
+	res := make([]byte, 8)
+	binary.BigEndian.PutUint64(res, uint64(c.Count))
+	return res, nil
+}
+
+func TestSerialModelBucketIndexScanUnique(t *testing.T) {
+	db := store.MemStore()
+
+	b := NewSerialModelBucket("cnts", &CounterWithID{}, WithIndexSerial("counter", lexographicCountIndex, true))
+
+	cnts := []CounterWithID{
+		CounterWithID{Count: 1},
+		CounterWithID{Count: 17},
+		CounterWithID{Count: 93},
```

Comment:
can you try this with >255 so the int is now multiple bytes in compressed form
this will show the different between varint encoding and the bigendian (fixed) encoding, as well as littleendian